cli/EAP7-924_Coloured_output_for_CLI.adoc

```diff
@@ -0,0 +1,83 @@
+= EAP7-924 Coloured output for CLI
+:author:            Ingo Weiss
+:email:             ingo@redhat.com
+:toc:               left
+:icons:             font
+:keywords:          comma,separated,tags
+:idprefix:
+:idseparator:       -
+:issue-base-url:    https://issues.jboss.org/browse/
+
+== Overview
+
+The intent of this feature is to provide a better user experience by means of colouring important parts of the CLI workflow, this includes error messages, warning messages, successful outputs and better visual feedback when using batch and flow control mechanisms if/for/try.
+
+== Issue Metadata
+
+=== Issue:
+
+* {issue-base-url}EAP7-924[EAP7-924]
+
+=== Related Issues:
+
+* {issue-base-url}WFCORE-3450
+* {issue-base-url}WFCORE-3577
+
+=== Dev Contacts:
+
+* mailto:ingo@redhat.com[Ingo Weiss]
+* mailto:jdenise@redhat.com[Jean-François Denise]
+
+=== QE Contacts:
+
+* mailto:eduda@redhat.com[Erich Duda]
+
+=== Affected Projects or Components:
+
+* WildFly Core CLI
+
+
+== Requirements
+
+=== Hard requirements
+
+* Ability to enable or disable colour output via CLI argument
+* Ability to enable or disable colour output using `jboss-cli.xml`
+* Ability to configure the colours for each allowed type: `warning`, `error`, `success`, `batch`, `required`
+* Mark required attributes in bold and with a different colour from the default
+* Colour the output of DMR output based on its outcome
+* Add colours to CLI prompt
+
+=== Soft requirements
+
+* Have a different prompt for each workflow step
+
+== Test Plan
+
+The following "matrix" should be tested on each WildFly/EAP supported operating system:
+
+
+[width="75%",options="header,footer"]
+|===
+| Terminal colours? | CLI colour
+| YES | ON
+| YES | OFF
+| NO  | ON
+| NO  | OFF
+|===
```

Comment:
I did a research how to emulate terminal which does not support colors, but I found nothing. I only found some ENV properties which can affect colors in command line applications, but it is up to the application to check them. It does not turn off the color capabilities in terminal.

As next step I tried to find real terminals without color capabilities. I tested the feature on all supported platforms included Solaris, HP-UX and Windows 2008, 2012 and 2016. All these platforms support colors in their terminals.

I also asked how this is tested in logging subsystem, but there are no tests which would test colors on terminals without color capabilities.

So the conclusion is that I don't know how to test this matrix.

| Terminal colours? | CLI colour | Test coverage  |
| ----------------------  | ------------- | ------------------- |
| YES | ON | Tested by tests developed for this RFE |
| YES | OFF | Tested by regression tests |
| NO | ON | Technically difficult to test. See aforementioned reasons. |
| NO | OFF | Technically difficult to test. See aforementioned reasons. |

I suggest to remove this testing requirement. Wdyt?

Reply:
+1 on removing it.

Reply:
I agree. If there's no platform to not support colours, no reason to push for it.

Reply:
I was going to merge this since the actual code got merged, but I see this bit is still here. Please remove it if unwanted, ping me on hipchat and i'll merge. Thanks!